process_report/tests/unit/invoices/test_pi_specific_invoice.py

```diff
@@ -1,68 +1,147 @@
 from unittest import TestCase, mock
 import tempfile
 import pandas
-import os
 
 from process_report.tests import util as test_utils
 
 
-class TestExportPICSV(TestCase):
-    def setUp(self):
-        data = {
-            "Invoice Month": ["2023-01", "2023-01", "2023-01", "2023-01", "2023-01"],
-            "Manager (PI)": ["PI1", "PI1", "PI1", "PI2", "PI2"],
-            "Institution": ["BU", "BU", "BU", "HU", "HU"],
-            "Project - Allocation": [
-                "ProjectA",
-                "ProjectB",
-                "ProjectC",
-                "ProjectD",
-                "ProjectE",
-            ],
-            "Untouch Data Column": ["DataA", "DataB", "DataC", "DataD", "DataE"],
-            "Is Billable": [True, True, True, True, True],
-            "Missing PI": [False, False, False, False, False],
-        }
-        self.dataframe = pandas.DataFrame(data)
-        self.invoice_month = data["Invoice Month"][0]
+class TestPISpecificInvoice(TestCase):
+    def _get_test_invoice(
+        self,
+        pi,
+        institution,
+        balance,
+        is_billable=None,
+        missing_pi=None,
+        group_name=None,
+    ):
+        if not is_billable:
+            is_billable = [True for _ in range(len(pi))]
 
-    @mock.patch("process_report.invoices.invoice.Invoice._filter_columns")
-    def test_export_pi(self, mock_filter_cols):
-        mock_filter_cols.return_value = self.dataframe
+        if not missing_pi:
+            missing_pi = [False for _ in range(len(pi))]
 
-        output_dir = tempfile.TemporaryDirectory()
-        pi_inv = test_utils.new_pi_specific_invoice(
-            output_dir.name, invoice_month=self.invoice_month, data=self.dataframe
+        if not group_name:
+            group_name = [None for _ in range(len(pi))]
```

Comment:
You can write instead:

```suggestion
            group_name = [None] * len(pi)
```

Compare:

```
>>> pi=[1,2,3,4,5]
>>> [None for _ in range(len(pi))]
[None, None, None, None, None]
>>> [None] * len(pi)
[None, None, None, None, None]
```